webpack.config.ts

```diff
@@ -11,28 +11,63 @@ import CopyWebpackPlugin from 'copy-webpack-plugin';
 import MiniCssExtractPlugin from 'mini-css-extract-plugin';
 import webpack, {Configuration} from 'webpack';
 
-function parseFeatureDetails(id: FeatureID): FeatureMeta {
-	const content = readFileSync(`source/features/${id}.tsx`, {encoding: 'utf-8'});
-	const fields = ['disabled', 'description', 'screenshot'] as const;
+const readmeContent = readFileSync('readme.md', 'utf-8');
+
+function stripLinks(markdownText: string): Partial<FeatureMeta> {
+	const urls: string[] = [];
+	const description = markdownText.replace(/\[(.+?)]\((.+?)\)/g, (_match, title, url) => {
+		urls.push(url);
+		return title;
+	});
+
+	return {
+		description,
+		screenshot: urls.find(url => url.startsWith('https://user-images.githubusercontent.com/')) ?? urls[0]
+	};
+}
 
+function parseFeatureDetails(id: FeatureID): FeatureMeta {
 	const feature: Partial<FeatureMeta> = {id};
-	for (const field of fields) {
-		const value = new RegExp(`\n\t${field}: '([^\\n]+)'`).exec(content)?.[1];
-		if (value) {
-			const validValue = value.trim().replace(/\\'/g, '’'); // Catch trailing spaces and incorrect apostrophes
-			if (value !== validValue) {
-				throw new Error(stripIndent(`
-					❌ Invalid characters found in \`${id}\`. Apply this patch:
-
-					- ${field}: '${value}'
-					+ ${field}: '${validValue}'
-				`));
-			}
 
-			feature[field] = value.replace(/\\\\/g, '\\');
+	const lineRegex = new RegExp(`^- \\[\\]\\(# "${id}"\\)(?: 🔥)? (.+)$`, 'm');
+	const lineMatch = lineRegex.exec(readmeContent);
+	if (lineMatch) {
+		Object.assign(feature, stripLinks(lineMatch[1]));
+		feature.description = feature.description!
+			.replace(/<code>\\`(.+?)\\`<\/code>/, '`$1`') // Simplify weird Markdown escaping
+			.replace(/<kbd>(.+?)<\/kbd>/g, '`$1`'); // Replace keyboard shortcut tags
+	} else {
+		// Feature might be highlighted in the readme
+		const imageRegex = new RegExp(`<img id="${id}" alt="(.+?)" src="(.+?)">`);
+		const imageMatch = imageRegex.exec(readmeContent);
+		if (imageMatch) {
+			feature.description = imageMatch[1];
+			feature.screenshot = imageMatch[2];
+		} else {
+			throw new Error(stripIndent(`
+				❌ Feature \`${id}\` needs a description in readme.md (please refer to the style guide there):
+
+				- [](# "${id}") [feature description](screenshot url)
+			`));
 		}
 	}
 
+	const validDescription = feature.description.trim().replace(/(?<!`)\\'/g, '’'); // Catch trailing spaces and incorrect apostrophes
+	if (feature.description !== validDescription) {
+		throw new Error(stripIndent(`
+			❌ Invalid characters found in description for \`${id}\`. Apply this patch:
+
+			- ${feature.description}'
+			+ ${validDescription}'
+		`));
+	}
+
+	const featureFileContent = readFileSync(`source/features/${id}.tsx`, {encoding: 'utf-8'});
+	const [, disabledReason] = /\n\tdisabled: '([^\n]+)'/.exec(featureFileContent) ?? [];
+	if (disabledReason) {
+		feature.disabled = disabledReason;
+	}
```

Comment:
How about this instead?

1. Change `getFeatures` to parse `refined-github.ts` instead
2. Disabled features will be _commented out_ in that file

The regex `^import.+?/([^.]+)\.tsx$` would include only non-commented-out JavaScript features

## Pros

- It guarantees that only imported features appear in `FeatureID[]`, just just _any file that is in `/features/`_
- Disabled features won't be bundled at all
- We can drop all code related to "disabled features"

## Cons

- When/if a feature is re-enabled, it won't remember if it was disabled (this happens so rarely that it doesn't matter much)